src/miner.cpp

```diff
@@ -440,21 +440,8 @@ bool CreateCoinStake( CBlock &blocknew, CKey &key,
     // Choose coins to use
     set <pair <const CWalletTx*,unsigned int> > CoinsToStake;
 
-    int64_t BalanceToStake = wallet.GetBalance();
-    int64_t nValueIn = 0;
-    //Request all the coins here, check reserve later
-
-    if (BalanceToStake <= 0) {
-        return BreakForNoCoins(MinerStatus, _("No coins"));
-    } else if (!wallet.SelectCoinsForStaking(BalanceToStake*2, txnew.nTime, CoinsToStake, nValueIn)) {
-        return BreakForNoCoins(MinerStatus, _("Waiting for coins to mature"));
-    }
-
-    BalanceToStake -= nReserveBalance;
-
-    if (BalanceToStake <= 0) {
-        return BreakForNoCoins(MinerStatus, _("Entire balance reserved"));
-    }
+    if (!wallet.SelectCoinsForStaking(txnew.nTime, CoinsToStake, true))
+        return false;
```

Comment:
ill double check this

Reply:
@iFoggz I think he is right about this.

Reply:
that is outdated.

its as this right now

```
   if (!wallet.SelectCoinsForStaking(txnew.nTime, CoinsToStake, sError, true))
    {
        ReturnMinerError(MinerStatus, sError);

        return false;
    }
```

Reply:
oh nvm thats in the miner let me check the GetStakeWEight

Reply:
ya i see in GetStakeWeight the nWeight should be set to 0 before that is called to work with new functionality. surprised stakeweight never showed otherwise then 0 on ui

Reply:
i've changed the location of nWeight so least when the weight is checked in ui/rpc if it fails either of those it'll be 0 returned as intended by original functionality of GetStakeWeight

Reply:
nWeight would of already been zero but good practice to make sure its 0 before checking SelectCoinsForStaking.

Reply:
@iFoggz I meant that if `vCoinsRet` is empty after calling `SelectCoinsForStaking()`, we need to update the miner status one last time with a reason similar to `_("No UTXOs satisfy reserve")`. This accounts for the situation where the balance is still greater than the reserve, but none of the UTXOs are small enough to stake without exceeding the reserve. Otherwise, the miner status message won't update--it'll show whatever reason was set before--but none of the coins can stake. 

I think we can set this reason in `SelectCoinsForStaking()` before shuffling the coins if `vCoinsRet.empty()`.

Reply:
yes, i see what u meant thought you were referring to vCoins. ill send back an error for that. ty